### PR TITLE
Fallback to min/max/step of Item if not set for Widget

### PR DESCRIPTION
**Pull request by mueller-ma**
Sliders require values to be Floats, so I changed
Item.minimum/maximum/step to Float.

Signed-off-by: mueller-ma <mueller-ma@users.noreply.github.com>

### CODE REVIEW

**Comment by mueller-ma** [comment]
@maniac103 FYI the new setpoint sliders won't cover the range from 0 to 100 with this change anymore, at least for my setup.

**Comment by maniac103** [comment]
What do you mean by 'won't cover'?

File: mobile/src/main/java/org/openhab/habdroid/model/Widget.kt [hunk 1]
```diff
@@ -285,9 +286,9 @@ fun JSONObject.collectWidgets(parent: Widget?): List<Widget> {
     val type = getString("type").toWidgetType()
     val icon = optStringOrNull("icon")
     val (minValue, maxValue, step) = Widget.sanitizeMinMaxStep(
-        optDouble("minValue", 0.0).toFloat(),
-        optDouble("maxValue", 100.0).toFloat(),
-        optDouble("step", 1.0).toFloat()
+        optFloatOrNull("minValue") ?: item?.minimum ?: 0f,
+        optFloatOrNull("maxValue") ?: item?.maximum ?: 100f,
+        optFloatOrNull("step") ?: item?.step ?: 1f
```
Comment (maniac103) [inline mobile/src/main/java/org/openhab/habdroid/model/Widget.kt:291]: Need to update `updateFromEvent` with this logic, or convert min/max/step to a getter function.

**Comment by mueller-ma** [comment]
> What do you mean by 'won't cover'?

I found this issue, because the new setpoint slider of my heating had the range 0°C - 100°C 😓

File: mobile/src/main/java/org/openhab/habdroid/model/Widget.kt (hunk 1 above)
Comment (mueller-ma) [reply mobile/src/main/java/org/openhab/habdroid/model/Widget.kt]: Done by adding getters.

**Pull request merged**